do-catalog/src/components/datasetDetail/DatasetHeader.vue

```diff
@@ -22,12 +22,50 @@
 
     <div class="u-ml--auto grid-cell grid-cell--col3 grid-cell--col4--tablet">
       <div class="u-flex u-flex__justify--end">
-        <Button v-if="isPublicWebsite" :url="getFormUrl()"
-          >I’m interested</Button
+        <Button
+          v-if="getSubscriptionStatus === 'interested'"
+          :url="getFormUrl()"
```

Comment:
I rather refactor all this ifs to something similar to
```
<Button @click.native="onSubscriptionButtonClicked()>
  {{ getSubscriptionButtonLabel }}
  <img src="getSubscriptionButtonIcon">
</Button>
```
And inside each computed/method set the needed logic.

Reply:
@neokore I can't do it like this because each button has different things apart from its label. For example, the unsubscribe button incorporates a span, other buttons change color, other has an icon, etc. Also, click events are different. I think this way is easier

Reply:
When setting computed/methods called `getSubscriptionButtonIcon` or `onSubscriptionButtonClicked` I meant to use this functions to control and return the different icons, actions, labels and even style classes. Also it is possible to add a span for all of the buttons and change their classes there or even set styles to the button and let its children inherit the needed styles to minify the number of needed functions.
The suggestion is just to avoid a large if-else chain in the template and make it possible to debug it and scale it easier in the future, though I know that maybe there are some more functions that it is desireable.
Anyway, we can cover it later.